utils/ticker.go

```diff
@@ -18,7 +18,7 @@ func SlowLogger(ctx context.Context, msg, fieldName, fieldVal string, logger log
 		for {
 			select {
 			case <-slowTicker.C:
-				elapsed := time.Since(startTime).String()
+				elapsed := time.Since(startTime).Round(time.Second).String()
```

Comment:
Using `Round(time.Second)` here means that 2.0001 and 2.002 are treated identically in terms of log deduplication ("2s"), which is fine IMO as this function really only cares about second-level granularity.